src/System.Private.ServiceModel/src/System/IdentityModel/Claims/X509CertificateClaimSet.cs

```diff
@@ -267,6 +268,32 @@ public override IEnumerable<Claim> FindClaims(string claimType, string right)
             }
         }
 
+        private static string[] GetDnsFromExtensions(X509Certificate2 cert)
+        {
+            foreach (X509Extension ext in cert.Extensions)
+            {
+                // Extension is SAN or SAN2
+                if (ext.Oid.Value == "2.5.29.7" || ext.Oid.Value == "2.5.29.17")
+                {
+                    string asnString = ext.Format(true);
+                    if (string.IsNullOrEmpty(asnString))
+                    {
+                        return new string[0];
+                    }
+
+                    string[] rawDnsEntries = asnString.Split(new string[1] { "\n" }, StringSplitOptions.RemoveEmptyEntries);
+                    string[] dnsEntries = new string[rawDnsEntries.Length];
+                    for (int i = 0; i < rawDnsEntries.Length; ++i)
+                    {
+                        int equalSignIndex = rawDnsEntries[i].IndexOf('=');
```

Comment:
There are a couple of problems with this.

* This won't work on x-plat, see https://github.com/dotnet/corefx/issues/3843 (aside: I missed moving this bug to github, glad it came up, so that I could link to it, and then couldn't find it!)
    * Firstly, because it doesn't have a way to print multiline, so the boolean doesn't do anything
    * Secondly, because it is "DNS:" as a prefix, vs Windows (en-US) "DNS Name="
* This is matching on all of the SAN entries.  You really only want the dNSName entries.  Sort of maybe you also want the iPAddress values, but those need to be matched as binary, not text (particularly IPv6, since there're a lot of text representations that are valid for localhost)

If not for the x-plat newlines, a viable option would be to save a byte\[\] of a Subject Alternative Names extension containing one value (and dNSName) and figure out what its prefix is; then match only on those entries.

(SAN Entries are each of type GeneralName:
```
   GeneralName ::= CHOICE {
        otherName                       [0]     OtherName,
        rfc822Name                      [1]     IA5String,
        dNSName                         [2]     IA5String,
        x400Address                     [3]     ORAddress,
        directoryName                   [4]     Name,
        ediPartyName                    [5]     EDIPartyName,
        uniformResourceIdentifier       [6]     IA5String,
        iPAddress                       [7]     OCTET STRING,
        registeredID                    [8]     OBJECT IDENTIFIER }
```
)